### PR TITLE
WIP: bugfix(): tag image with <image-name> from build spec and push to registry/registries

### PR DESCRIPTION
**Pull request by AniketKul**
This PR fixes https://github.com/ocibuilder/ocibuilder/issues/70

### CODE REVIEW

**Comment by AniketKul** [comment]
Getting 

```
DEBU[2019-11-25 09:50:32] found custom params in spec.yaml              number=2
DEBU[2019-11-25 09:50:32] setting param value at destination            dest=login.0.this.path.is.wrong value=testuser
ERRO[2019-11-25 09:50:32] Error validating params, check that your param dest is valid  dest=login.0.this.path.is.wrong error="path to dest is invalid in a set param"
```

from master branch. working on fixing this test case.

**Comment by AniketKul** [comment]
that test is supposed to fail. All good. Just need to fix one unused parameter error in clientset

**Comment by AniketKul** [comment]
docker cli converts `docker.io/<user>/<image-name>:<tag>` to `<user>/<image-name>:<tag>` internally and I had a bug in my code so e2e was failing. 

```
go run main.go 
failed to tag image before pushingpanic: Error response from daemon: No such image: lol:latest

goroutine 1 [running]:
```

that bug is fixed now. Checking further

**Comment by AniketKul** [comment]
Good to merge now!

**Comment by AniketKul** [comment]
will resolve the conflicts with master

**Comment by AniketKul** [comment]
this one needs re-work too since the recent merge to master got rid of `pkg/docker/docker.go` and `pkg/buildah/buildah.go` files

File: pkg/docker/docker.go [hunk 1]
```diff
@@ -178,18 +178,33 @@ func (d Docker) Push(spec v1alpha1.OCIBuilderSpec) ([]io.ReadCloser, error) {
...
+
+	for i, pushSpec := range spec.Push {
 		if err := common.ValidatePushSpec(&pushSpec); err != nil {
 			return nil, err
 		}
 
-		pushImageName := fmt.Sprintf("%s/%s:%s", pushSpec.Registry, pushSpec.Image, pushSpec.Tag)
-		log.WithField("name:", pushImageName).Infoln("pushing image")
+		pushImageName := fmt.Sprintf("%s/%s/%s:%s", pushSpec.Registry, pushSpec.User, pushSpec.Image, pushSpec.Tag)
+		log.Infoln("Ozark1: ", pushImageName)
```
[the remark this answers is not in the document]
Comment (AniketKul) [reply pkg/docker/docker.go]: already removed

**Comment by AniketKul** [comment]
getting a couple of `DeepCopy` and `DeepCopInto` errors with newly generated `zz_generated.deepcopy.go` on master. working on fixing that.

**Comment by AniketKul** [comment]
> Based on discussion on call - we want to combine registry, user and image into a single field and be consistent with dockers naming for this new field

Okay.

**Comment by artbegolli** [comment]
@AniketKul will be creating a new PR for this as well?

**Comment by AniketKul** [comment]
@artbegolli - yes I will create a new PR for this one. With the new changes, I think it makes sense to get my changes merged into the newly merged changes from master branch

**Comment by artbegolli** [comment]
> @artbegolli - yes I will create a new PR for this one. With the new changes, I think it makes sense to get my changes merged into the newly merged changes from master branch

👍 gonna close this then

**Pull request closed without merging**